includes/sanitizers/class-amp-base-sanitizer.php

```diff
@@ -661,4 +699,53 @@ public function maybe_add_amp_image_lightbox_node() {
 		);
 		$body_node->appendChild( $amp_image_lightbox );
 	}
+
+	/**
+	 * Parse a style string into an associative array of style attributes.
+	 *
+	 * @param string $style_string Style string to parse.
+	 * @return array Associative array of style attributes.
+	 */
+	protected function parse_style_string( $style_string ) {
```

Comment:
Beautiful function. 😍

Reply:
I don't trust it and am not sure this is a good way to go about. I would prefer to use actual parsing like with PHP-CSS-Parser, but that seems to only work through a hack, too.

Reply:
I don't think it is actually a problem here, as we're only looking for 2 very specific patterns, and not using malformed CSS in any other way anyway, so we're good for now. But the function shouldn't be used for anything more involved than that.

Reply:
Yes, better to not use the PHP-CSS-Parser because it is too expensive for this runtime operation.